arduino-ide-extension/src/node/sketches-service-impl.ts

```diff
@@ -658,11 +658,24 @@ export class SketchesServiceImpl
     return FileUri.create(genBuildPath).toString();
   }
 
-  async getIdeTempFolderPath(sketch: Sketch): Promise<string> {
+  private async getIdeTempFolderPath(sketch: Sketch): Promise<string> {
     const sketchPath = FileUri.fsPath(sketch.uri);
     await fs.readdir(sketchPath); // Validates the sketch folder and rejects if not accessible.
     const suffix = crypto.createHash('md5').update(sketchPath).digest('hex');
-    return path.join(os.tmpdir(), `arduino-ide2-${suffix}`);
+    return path.join(
+      this.isTempSketch.tempDirRealpath,
+      `arduino-ide2-${suffix}`
+    );
+  }
+
+  async tempBuildPath(sketch: Sketch): Promise<string> {
+    const sketchPath = FileUri.fsPath(sketch.uri);
+    const hash = crypto
+      .createHash('md5')
+      .update(sketchPath)
+      .digest('hex')
+      .toUpperCase();
+    return join(this.isTempSketch.tempDirRealpath, `arduino-sketch-${hash}`);
```

Comment:
When I follow the reproduction instructions from https://github.com/arduino/arduino-ide/issues/808#issue-1123946833 on my Windows machine, I get the raw error from Arduino CLI in the notification:

> Error getting Debug info: Compiled sketch not found in C:\Users\per\AppData\Local\Temp\arduino-sketch-AEA9FCC61D66FCE9E9A157F96547FFB0

Instead of the expected user friendly interpretation from Arduino IDE:

> Sketch 'sketch_jan16b' must be verified before starting a debug session. Please verify the sketch and start debugging again. Do you want to verify the sketch now?

I set a breakpoint here:

https://github.com/arduino/arduino-ide/blob/2095e72d0ccb43a12ea2ef5e59fbcfa7b276b52f/arduino-ide-extension/src/browser/contributions/debug.ts#L250

I see that the value of `tempBuildPath` is:

`c:\\Users\\per\\AppData\\Local\\Temp\\arduino-sketch-956C36A70086EB62B81F7E1AA8AEE643`

In addition to the difference in hash, I notice the different case of the drive letters (`c` vs `C`).

---

I get the expected user friendly notification text on my Linux machine.

Reply:
Thanks for the review. I patched the PR. It's a complete hack, but finally it works on Windows too:


https://user-images.githubusercontent.com/1405703/212833590-9c1507ae-1eaa-4953-876e-91c46e57d365.mp4